container/options.go

```diff
@@ -60,6 +59,43 @@ func validateOptions(c *Container) error {
 	return nil
 }
 
+// ensure all the container only have one split modifier.
+func validateSplits(c *Container) error {
+	var errStr string
+	preOrder(c, &errStr, func(c *Container) error {
+		if c.opts.splitFixed > -1 && c.opts.splitPercent != 50 {
+			return fmt.Errorf(
+				"only one of splitFixed `%v` and splitPercent `%v` is allowed to be set per container",
+				c.opts.splitFixed,
+				c.opts.splitPercent,
+			)
+		}
+
+		return nil
+	})
+	if errStr != "" {
+		return errors.New(errStr)
+	}
+	return nil
+}
+
+// validateOptions validates options set in the container tree.
+func validateOptions(c *Container) error {
+	// ensure all the container identifiers are either empty or unique.
+	errIds := validateIds(c)
```

Comment:
(optional) If we want to we could reduce the multiple traversal issue by calling preOrder just once from within validateOptions().

Related to an earlier comment.